LandSandBoat/server
# [Core] Rework getSpeed()

**_I affirm:_**

## What does this pull request do?

Tries to get us closer to matching what retail does with speed in the following ways:
- Creates a set of modifiers to separate movement speed sources.
- Makes possitive movement speed bonuses from gear NOT stack with each other and take the highest only instead.
- Makes Mazurka and quickening not stack and take the highest of the two.
- Makes weight and curse, among others, stack with each other and be multiplicative to other sources.
- Fixes an issue with !speed GM command when using movement gear.

Resolves #2482, resolves #2499, resolves #2501 

## Retail observations:
- Positive movement speed from gear DOES NOT stack with each other. Only highest applies.
- Negative movement speed from gear DOES stack with each other.
- Flee puts you at 80 speed. 2 pieces of dusk gear did NOT affect movement speed. The third one did. We concluded flee gives a bonus higher than the cap.
- Max speed is 80. There is no possible way that we could come up a different combination which allowed us to surpass this limit.
  - Combination -> +18% speed ring + Chocobo mazurka + sprinter shoes useable item effect + PERFECT Bolters roll + KI resulted in flee speed. 
- Positive movement speed from gear resulted in diminishing returns when combined with negative gear.
  - Bloody pants + dusk mask should result in ABOUT the same speed as regular speed. However, it was notizably slower, just not THAT much.
- Weight effects are concluded to be multiplicative. Full dusk gear + mollusca mantle + weight + curse resulted in a very slow movement speed, but not much different than just gear. If it was additive, we would have just stoped.
  - This was not the case of a lower limit cap. Beadeaux curse + weight is slower than just curse. Similarly to positive movement speed from gear, we saw diminishing returns (in this case, a diminishing penalty, since just curse with no negative gear is still very notizable)

## Steps to test these changes

Go to retail and check the actions described above.
Perform those same actions here and see if its at the very least SOMEWHAT close.
